infra/src/main/java/com/redhat/service/bridge/infra/k8s/KubernetesClient.java

```diff
@@ -0,0 +1,24 @@
+package com.redhat.service.bridge.infra.k8s;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.networking.v1.Ingress;
+
+public interface KubernetesClient {
+    void createOrUpdateDeployment(Deployment deployment);
+
+    void deleteDeployment(String name);
+
+    Deployment getDeployment(String name);
+
+    void createOrUpdateCustomResource(String name, Object customResource, String type);
```

Comment:
If `Object` here is instead `BridgeCustomResource` and that has a `getType()` method on it, does that simplify things? `BridgeCustomResource` and `ProcessorCustomResource` would then extend `BridgeCustomResource`.

I see we have method calls where we do things like `createOrUpdateCustomResource("foo, BridgeCustomResource.fromDTO(bridgeDTO), K8SBridgeConstants.BRIDGE_TYPE);`

I can easily do something like `createOrUpdateCustomResource("foo, BridgeCustomResource.fromDTO(bridgeDTO), K8SBridgeConstants.PROCESSOR_TYPE);`, whereas if we have an `BridgeCustomResource`, I cannot do that.